src/transformers/models/musicgen_melody/modeling_musicgen_melody.py

```diff
@@ -2428,6 +2451,16 @@ def _maybe_initialize_input_ids_for_generation(
                 break
         return torch.ones((batch_size, 1), dtype=torch.long, device=self.device) * bos_token_id
 
+    def freeze_encoders(self, freeze_text_encoder=True):
```

Comment:
You can also do a simple test to confirm this method works as expected: https://github.com/huggingface/transformers/blob/440bd3c3c06004115752c57530c7150b38f2a0a7/tests/models/whisper/test_modeling_whisper.py#L435